engine/class_modules/priest/sc_priest.hpp

```diff
@@ -955,7 +945,15 @@ struct fiend_melee_t : public priest_pet_melee_t
     if ( base_execute_time == timespan_t::zero() )
       return timespan_t::zero();
 
-    return base_execute_time * player->cache.spell_speed();
+    // Mindbender inherits haste from the player
+    timespan_t hasted_time = base_execute_time * player->cache.spell_speed();
+
+    if ( p().o().conduits.rabid_shadows->ok() )
```

Comment:
If you want to be consistent you can just use .ok() here as well instead of doing -> indirection. But not really relevant.